src/main/sensors/esc_sensor.c

```diff
@@ -361,8 +361,10 @@ void escSensorProcess(timeUs_t currentTimeUs)
     }
 }
 
+#endif
+
 int calcEscRpm(int erpm)
 {
-    return (erpm * 100) / (motorConfig()->motorPoleCount / 2);
+    //return (erpm * 100) / (motorConfig()->motorPoleCount / 2);
+    return (erpm * 200) / motorConfig()->motorPoleCount;
```

Comment:
```suggestion
    return lrintf(erpm * 200.0f / motorConfig()->motorPoleCount);
```

Just noticed that we can get more precision by using float division and then rounding to the nearest integer.

Reply:
I thing that as erpm is first multiplied by 200 and then divided by pole count there is no gain in precision, because in the end the value calculated is converter to an int. Maybe the value would change during rounding operation. For printing it in the osd I think this is no relevant, and for RPM filtering I think the filter would be wide enough. Also using floats and more conversions result in more overload and flash used. I  would prefer to leave as is, but please let me know your opinion, and if you want I will change it.

Reply:
If you really want to use rounding, you can use `(erpm * 200 + motorConfig()->motorPoleCount / 2 ) / motorConfig()->motorPoleCount`
(this way of rounding works only for positive values, but it is possible to extend it)

Reply:
My intention here is to use simpler calculations in order to max performance, minimize flash memory footprint and simplify code. Please, could you let me know what is the advantage of using that formula?

Reply:
The calculation is done using integers, so conversion to floating point, rounding and conversion back is not necessary. Code size increases by about 2 instruction ( shr + add)

Reply:
I understand, but please what is the advantage over the formula proposed?
```
return (erpm * 200) / motorConfig()->motorPoleCount;
```

Reply:
Sorry, got lost a bit. First comment was to @KarateBrot suggestion. May problem was with floating point math.

Difference is negligible in all real-world cases.

Reply:
I disagree. We can also use this function for other parts in the code where we need as much precision as possible. Without the suggestion the RPM numbers can be off by 13RPM (with 14 motor poles). With the suggestion we can only be off by 0.5RPM max.

We have potato MCU's but they are not THAT slow. We don't need to sacrifice performance for negligible speed increases.

Reply:
Please, could you show an example of a calculation where le loss of precission is so high as you mentioned?

Reply:
Ok, lets say 70 erpm so
(70 * 200) / 14 = 14000 / 14 = 1000 rpm

Or 44 erpm
(44 * 200) / 14 = 8800 / 14 = 628,57 rpm (loss of 0,57 rounding to int)

Or 456 erpm
(456 * 200) / 14 = 91200 / 14 = 6514,28 (loss of 0,28 rounding to int)

I think that the loss of precision is allways < 1.
Why do you say the loss of precission is so high?

Reply:
I understand. But here the problem is that erpm always come from the esc as an integer number, never as a float, so it has no effect working with floats.

Reply:
Ok sorry guys, you are right. My brain was playing a trick on me. I thought about the modulo for whatever reason the whole time 🤦‍♂️

Reply:
@damosvil : There is some effect, but it is negligible in this case. In general case, @KarateBrot would be right (you can get integer that is 0.5 closer to true value than truncating division)

(sorry, just being too pedantic)

Reply:
If you really want more precision, we need to update how many units the erpm is given to us from bidirectional dshot. Currently our erpm reading is really erpm*100. Because of that if you have a step change of 1 erpm with a 14 pole motor you end up with 100 / (14/2) which is 14.2857. So the most accurate we can be is to 14hz as we use an integer to represent the rpm. Even doubling the precision of the erpm would help as we would have 7.1428 per erpm we receive.

Reply:
What are thoughts on changing this line of code to increase accuracy? https://github.com/betaflight/betaflight/blob/025ee87a7aca068e3659fd066b8a9afbed123361/src/main/drivers/dshot_bitbang_decode.c#L85

Reply:
I would prefer to fix that in a different pr

Reply:
@Quick-Flash I think that that formula cannot be improved, because compiler reduces automatically constant part so
```
value = (1000000 * 60 / 100 + value / 2) / value;
```

is converted to
```
value = (600000 + value / 2) / value;
```

But in this case I don't see how it can be reduced to a simpler formula.

Sorry for the late explanation.